src/charm.py

```diff
@@ -377,6 +570,27 @@ def configure_ceph(self, event) -> None:
                 return
             raise e
 
+    def _update_service_endpoints(self):
+        try:
+            if self.id_svc.update_service_endpoints:
+                logger.debug("Updating service endpoints after ingress relation changed")
+                self.id_svc.update_service_endpoints(self.service_endpoints)
+        except (AttributeError, KeyError):
+            pass
```

Comment:
I'm a bit wary that we might be hiding faults in update_service_endpoints() by silently dropping an except here.
May I suggest to separate checking for existence of self.id_svc.update_service_endpoints from actual calling?

Reply:
The attribute error can be raised if `self.id_svc.update_service_endpoints` does not exist (which should not be the case irrespective of relation joined or not) OR from `self.id_svc.update_service_endpoints(self.service_endpoints)` when the relation is not integrated.
In both the cases, the action should be ignore. So i added logging to print the error message. Any other kind of exceptions/errors in `self.id_svc.update_service_endpoints(self.service_endpoints)` results charm in blocked state.